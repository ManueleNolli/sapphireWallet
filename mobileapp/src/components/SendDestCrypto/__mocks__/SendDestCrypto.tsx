const SendDestCryptoMock = () => <div data-testid="SendDestCrypto">SendDestCrypto</div>

export default SendDestCryptoMock
