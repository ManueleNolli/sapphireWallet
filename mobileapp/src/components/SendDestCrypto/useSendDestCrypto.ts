import { useContext, useState } from 'react'
import { BlockchainContext } from '../../context/BlockchainContext'
import { WalletContext } from '../../context/WalletContext'
import { getSigner } from '../../services/wallet'
import Toast from 'react-native-toast-message'
import useLoading from '../../hooks/useLoading'
import { Signer } from 'ethers'
import { NETWORKS } from '../../constants/Networks'
import { BRIDGE_NETWORKS } from '../../constants/BridgeNetworks'
import { executeTransactionResponse } from '../../services/backend'

type useSendDestCryptoProps = {
  address: string
  cryptoName: string
  action: (
    walletAddress: string,
    to: string,
    value: number,
    signer: Signer,
    network: NETWORKS,
    destinationNetwork: BRIDGE_NETWORKS,
    internalSapphireTX: boolean
  ) => Promise<executeTransactionResponse>
  close: (needRefresh: boolean) => void
}
export default function useSendDestCrypto({ address, cryptoName, action, close }: useSendDestCryptoProps) {
  const { getPrivateKey } = useContext(WalletContext)
  const { currentNetwork } = useContext(BlockchainContext)
  const { isLoading, setIsLoading } = useLoading()
  const [valueAddress, setValueAddress] = useState<string>('')
  const [isAddressValid, setIsAddressValid] = useState<boolean>(false)
  const [valueAmount, setValueAmount] = useState<string>('')
  const [isAmountValid, setIsAmountValid] = useState<boolean>(false)
  const [checkedIsSapphireInternalTX, setCheckedIsSapphireInternalTX] = useState<boolean>(true)
  const [isQRCodeScanning, setIsQRCodeScanning] = useState<boolean>(false)

  const sendDestCryptoTransaction = async () => {
    setIsLoading(true)
    const value = Number.parseFloat(valueAmount)
    const signer = await getSigner(await getPrivateKey(`Sign transaction to send ${cryptoName}.`), currentNetwork)

    try {
      await action(
        address,
        valueAddress,
        value,
        signer,
        currentNetwork,
        BRIDGE_NETWORKS.AMOY,
        checkedIsSapphireInternalTX
      )
      setIsLoading(false)
      close(true)
      Toast.show({
        type: 'success',
        text1: 'Transaction sent! ðŸš€',
      })
    } catch (e: any) {
      Toast.show({
        type: 'error',
        text1: 'Transaction failed! ðŸ˜¢',
        text2: e.message,
      })
      setIsLoading(false)
      close(false)
    }
  }

  const QRCodeFinishedScanning = (data: string) => {
    setValueAddress(data)
    setIsQRCodeScanning(false)
  }

  return {
    isLoading,
    sendDestCryptoTransaction,
    valueAddress,
    setValueAddress,
    isAddressValid,
    setIsAddressValid,
    valueAmount,
    setValueAmount,
    isAmountValid,
    setIsAmountValid,
    isQRCodeScanning,
    setIsQRCodeScanning,
    QRCodeFinishedScanning,
    checkedIsSapphireInternalTX,
    setCheckedIsSapphireInternalTX,
  }
}
