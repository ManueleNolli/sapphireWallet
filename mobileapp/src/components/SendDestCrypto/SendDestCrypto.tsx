import React from 'react'
import { Button, CheckBox, Layout, Spinner, StyleService, Text, useStyleSheet } from '@ui-kitten/components'
import { vh, vw } from '../../Styles'
import useSendDestCrypto from './useSendDestCrypto'
import InputAddress from '../InputAddress/InputAddress'
import InputNumeric from '../InputNumeric/InputNumeric'
import { View } from 'react-native'
import QRCodeScanner from '../QRCodeScanner/QRCodeScanner'
import { Signer } from 'ethers'
import { NETWORKS } from '../../constants/Networks'
import { BRIDGE_NETWORKS } from '../../constants/BridgeNetworks'
import { executeTransactionResponse } from '../../services/backend'

type SendDestCryptoProps = {
  address: string
  cryptoName: string
  balance: number
  action: (
    walletAddress: string,
    to: string,
    value: number,
    signer: Signer,
    network: NETWORKS,
    destinationNetwork: BRIDGE_NETWORKS,
    internalSapphireTX: boolean
  ) => Promise<executeTransactionResponse>
  close: (needRefresh: boolean) => void
}
export default function SendDestCrypto({ address, cryptoName, balance, action, close }: SendDestCryptoProps) {
  const styles = useStyleSheet(themedStyles)
  const {
    isLoading,
    sendDestCryptoTransaction,
    valueAddress,
    setValueAddress,
    isAddressValid,
    setIsAddressValid,
    valueAmount,
    setValueAmount,
    isAmountValid,
    setIsAmountValid,
    isQRCodeScanning,
    setIsQRCodeScanning,
    QRCodeFinishedScanning,
    checkedIsSapphireInternalTX,
    setCheckedIsSapphireInternalTX,
  } = useSendDestCrypto({
    address,
    cryptoName,
    action,
    close,
  })

  const LoadingIndicator = (props: any) => {
    return (
      isLoading && (
        <View style={[props.style, styles.indicator]}>
          <Spinner size="tiny" />
        </View>
      )
    )
  }

  if (isQRCodeScanning) return <QRCodeScanner onQRCodeScanned={QRCodeFinishedScanning} />

  return (
    <Layout style={styles.container}>
      <Text category="h6">Send {cryptoName}</Text>
      <Text style={{ marginBottom: 2 * vh }}>Fill the form to send {cryptoName} to another wallet</Text>
      <InputAddress
        label={"to"}
        value={valueAddress}
        setValue={setValueAddress}
        isValid={isAddressValid}
        setIsValid={setIsAddressValid}
        setIsQRCodeScanning={setIsQRCodeScanning}
      />
      <InputNumeric
        value={valueAmount}
        setValue={setValueAmount}
        isValid={isAmountValid}
        setIsValid={setIsAmountValid}
        maxValue={balance}
        style={{ marginTop: 1 * vh }}
      />
      <CheckBox
        testID="checkbox-button"
        status="danger"
        checked={checkedIsSapphireInternalTX}
        onChange={(nextChecked) => setCheckedIsSapphireInternalTX(nextChecked)}
        style={styles.checkbox}
      >
        {(evaProps) => (
          <Text category="label" style={{ marginLeft: 2 * vw }} status="danger">
            Receiver is a Sapphire wallet
          </Text>
        )}
      </CheckBox>
      <Button
        testID="send-button"
        accessoryRight={<LoadingIndicator />}
        style={{ marginTop: 2 * vh, width: '100%' }}
        appearance="outline"
        status="info"
        disabled={!(isAddressValid && isAmountValid)}
        onPress={sendDestCryptoTransaction}
      >
        {`Send ${cryptoName}`}
      </Button>
    </Layout>
  )
}

const themedStyles = StyleService.create({
  container: {
    paddingVertical: 4 * vh,
    paddingHorizontal: 4 * vw,
    alignItems: 'center',
    borderRadius: 10,
    maxHeight: 75 * vh,
    maxWidth: 90 * vw,
  },
  indicator: {
    position: 'absolute',
    right: 5,
    alignItems: 'center',
  },
  checkbox: {
    alignSelf: 'flex-start',
    marginTop: 2 * vh,
  },
})
