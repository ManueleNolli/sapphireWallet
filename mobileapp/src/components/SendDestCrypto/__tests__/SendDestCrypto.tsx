import useSendDestCrypto from '../useSendDestCrypto'
import renderWithTheme from '../../../TestHelper'
import React from 'react'

import { act, fireEvent, waitFor } from '@testing-library/react-native'
import SendDestCrypto from '../SendDestCrypto'

jest.mock('../useSendDestCrypto', () => jest.fn())
jest.mock('../../InputAddress/InputAddress')
jest.mock('../../InputNumeric/InputNumeric')
jest.mock('../../QRCodeScanner/QRCodeScanner')
jest.mock('@ui-kitten/components', () => {
  const { View } = require('react-native')
  return {
    ...jest.requireActual('@ui-kitten/components'),
    Spinner: () => <View data-testid="spinner" />,
  }
})

describe('SendDestCrypto', () => {
  it('renders correctly', async () => {
    ;(useSendDestCrypto as jest.Mock).mockReturnValue({
      isLoading: false,
      sendDestCryptoTransaction: jest.fn(),
      valueAddress: '0x123456789',
      setValueAddress: jest.fn(),
      isAddressValid: true,
      setIsAddressValid: jest.fn(),
      valueAmount: '100',
      setValueAmount: jest.fn(),
      isAmountValid: true,
      setIsAmountValid: jest.fn(),
      checkedIsSapphireInternalTX: true,
      setCheckedIsSapphireInternalTX: jest.fn(),
    })

    let tree: any

    await waitFor(async () => {
      tree = renderWithTheme(
        <SendDestCrypto
          address="0x123456789"
          cryptoName="NameCrypto"
          balance={100}
          close={jest.fn()}
          action={jest.fn()}
        />
      )
    })

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders correctly when loading', async () => {
    ;(useSendDestCrypto as jest.Mock).mockReturnValue({
      isLoading: true,
      sendDestCryptoTransaction: jest.fn(),
      valueAddress: '0x123456789',
      setValueAddress: jest.fn(),
      isAddressValid: true,
      setIsAddressValid: jest.fn(),
      valueAmount: '100',
      setValueAmount: jest.fn(),
      isAmountValid: true,
      setIsAmountValid: jest.fn(),
      checkedIsSapphireInternalTX: true,
      setCheckedIsSapphireInternalTX: jest.fn(),
    })

    let tree: any

    await waitFor(async () => {
      tree = renderWithTheme(
        <SendDestCrypto
          address="0x123456789"
          cryptoName="NameCrypto"
          balance={100}
          close={jest.fn()}
          action={jest.fn()}
        />
      )
    })
    expect(tree).toMatchSnapshot()
  })

  it('Send Button call sendETHTransaction', async () => {
    const sendMock = jest.fn()
    ;(useSendDestCrypto as jest.Mock).mockReturnValue({
      isLoading: true,
      sendDestCryptoTransaction: sendMock,
      valueAddress: '0x123456789',
      setValueAddress: jest.fn(),
      isAddressValid: true,
      setIsAddressValid: jest.fn(),
      valueAmount: '100',
      setValueAmount: jest.fn(),
      isAmountValid: true,
      setIsAmountValid: jest.fn(),
      checkedIsSapphireInternalTX: true,
      setCheckedIsSapphireInternalTX: jest.fn(),
    })

    let tree: any
    await waitFor(async () => {
      tree = renderWithTheme(
        <SendDestCrypto
          address="0x123456789"
          cryptoName="NameCrypto"
          balance={100}
          close={jest.fn()}
          action={jest.fn()}
        />
      )
    })

    await act(async () => {
      const button = tree.getByTestId('send-button')
      fireEvent.press(button)
    })

    expect(sendMock).toHaveBeenCalled()
  })

  it('Change CheckBox status', async () => {
    const checkedMock = jest.fn()

    ;(useSendDestCrypto as jest.Mock).mockReturnValue({
      isLoading: true,
      sendDestCryptoTransaction: jest.fn(),
      valueAddress: '0x123456789',
      setValueAddress: jest.fn(),
      isAddressValid: true,
      setIsAddressValid: jest.fn(),
      valueAmount: '100',
      setValueAmount: jest.fn(),
      isAmountValid: true,
      setIsAmountValid: jest.fn(),
      checkedIsSapphireInternalTX: true,
      setCheckedIsSapphireInternalTX: checkedMock,
    })

    let tree: any
    await waitFor(async () => {
      tree = renderWithTheme(
        <SendDestCrypto
          address="0x123456789"
          cryptoName="NameCrypto"
          balance={100}
          close={jest.fn()}
          action={jest.fn()}
        />
      )
    })

    await act(async () => {
      const button = tree.getByTestId('checkbox-button')
      fireEvent.press(button)
    })

    expect(checkedMock).toHaveBeenCalled()
  })

  it('Show QRCode Scanner', async () => {
    ;(useSendDestCrypto as jest.Mock).mockReturnValue({
      isLoading: true,
      sendDestCryptoTransaction: jest.fn(),
      valueAddress: '0x123456789',
      setValueAddress: jest.fn(),
      isAddressValid: true,
      setIsAddressValid: jest.fn(),
      valueAmount: '100',
      setValueAmount: jest.fn(),
      isAmountValid: true,
      setIsAmountValid: jest.fn(),
      isQRCodeScanning: true,
      checkedIsSapphireInternalTX: true,
      setCheckedIsSapphireInternalTX: jest.fn(),
    })

    let tree: any

    await waitFor(async () => {
      tree = renderWithTheme(
        <SendDestCrypto
          address="0x123456789"
          cryptoName="NameCrypto"
          balance={100}
          close={jest.fn()}
          action={jest.fn()}
        />
      )
    })

    expect(tree).toMatchSnapshot()
  })
})
