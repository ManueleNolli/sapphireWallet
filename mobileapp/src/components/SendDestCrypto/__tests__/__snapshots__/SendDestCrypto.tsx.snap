// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendDestCrypto Show QRCode Scanner 1`] = `
<div
  data-testid="QRCodeScanner"
>
  QRCodeScanner
</div>
`;

exports[`SendDestCrypto renders correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1000.5,
        "maxWidth": 675,
        "paddingHorizontal": 30,
        "paddingVertical": 53.36,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        undefined,
      ]
    }
  >
    Send 
    NameCrypto
  </Text>
  <Text
    appearance="default"
    category="p1"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
        },
        {
          "marginBottom": 26.68,
        },
      ]
    }
  >
    Fill the form to send 
    NameCrypto
     to another wallet
  </Text>
  <div
    data-testid="InputAddress"
  >
    InputAddress
  </div>
  <div
    data-testid="InputNumeric"
  >
    InputNumeric
  </div>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "marginTop": 26.68,
        "opacity": 1,
      }
    }
    testID="checkbox-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderRadius": 6,
              "height": 32,
              "width": 32,
            },
            {
              "position": "absolute",
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#FF3D71",
              "borderColor": "#FF3D71",
              "borderRadius": 3,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="12"
          bbWidth="12"
          fill="#FFFFFF"
          focusable={false}
          height={12}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#FFFFFF"
          strokeWidth={3}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 12,
                "width": 12,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={12}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeWidth="3"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGRect
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  height="24"
                  opacity={0}
                  width="24"
                  x="0"
                  y="0"
                />
                <RNSVGPath
                  d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      appearance="default"
      category="label"
      status="danger"
      style={
        [
          {
            "color": "#FF3D71",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "800",
          },
          {
            "marginLeft": 15,
          },
        ]
      }
    >
      Receiver is a Sapphire wallet
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 149, 255, 0.08)",
        "borderColor": "#0095FF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 26.68,
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
    testID="send-button"
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "color": "#0095FF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Send NameCrypto
    </Text>
  </View>
</View>
`;

exports[`SendDestCrypto renders correctly when loading 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1000.5,
        "maxWidth": 675,
        "paddingHorizontal": 30,
        "paddingVertical": 53.36,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        undefined,
      ]
    }
  >
    Send 
    NameCrypto
  </Text>
  <Text
    appearance="default"
    category="p1"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
        },
        {
          "marginBottom": 26.68,
        },
      ]
    }
  >
    Fill the form to send 
    NameCrypto
     to another wallet
  </Text>
  <div
    data-testid="InputAddress"
  >
    InputAddress
  </div>
  <div
    data-testid="InputNumeric"
  >
    InputNumeric
  </div>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "marginTop": 26.68,
        "opacity": 1,
      }
    }
    testID="checkbox-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderRadius": 6,
              "height": 32,
              "width": 32,
            },
            {
              "position": "absolute",
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#FF3D71",
              "borderColor": "#FF3D71",
              "borderRadius": 3,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            },
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="12"
          bbWidth="12"
          fill="#FFFFFF"
          focusable={false}
          height={12}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#FFFFFF"
          strokeWidth={3}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 12,
                "width": 12,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={12}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeWidth="3"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGRect
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  height="24"
                  opacity={0}
                  width="24"
                  x="0"
                  y="0"
                />
                <RNSVGPath
                  d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      appearance="default"
      category="label"
      status="danger"
      style={
        [
          {
            "color": "#FF3D71",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "800",
          },
          {
            "marginLeft": 15,
          },
        ]
      }
    >
      Receiver is a Sapphire wallet
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 149, 255, 0.08)",
        "borderColor": "#0095FF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 26.68,
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
    testID="send-button"
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "color": "#0095FF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Send NameCrypto
    </Text>
    <View
      style={
        [
          {
            "height": 20,
            "marginHorizontal": 10,
            "tintColor": "#0095FF",
            "width": 20,
          },
          {
            "alignItems": "center",
            "position": "absolute",
            "right": 5,
          },
        ]
      }
    >
      <View
        data-testid="spinner"
      />
    </View>
  </View>
</View>
`;
